File: pyround1/167.两数之和-ii-输入有序数组.py
#
# @lc app=leetcode.cn id=167 lang=python
#
# [167] 两数之和 II - 输入有序数组
#

# @lc code=start
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        if not numbers:
            return []
        res = []
        n = len(numbers)
        l = 0
        r = n-1
        while(l < r):
            if (numbers[l] + numbers[r] == target):
                return [l+1, r+1]
            elif(numbers[l] + numbers[r] > target):
                r = r-1
            else:
                l = l+1
        return [-1, -1]
# @lc code=end

File: 167.两数之和-ii-输入有序数组.cpp
/*
 * @lc app=leetcode.cn id=167 lang=cpp
 *
 * [167] 两数之和 II - 输入有序数组
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        unordered_map<int, int> map;
        for(int i = 0; i < numbers.size(); i++){
            if(map[target-numbers[i]])
            {
                return {map[target - numbers[i]], i+1};
            }
            map[numbers[i]] = i+1;
        }
        return {};

    }
};
// @lc code=end

